modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpRequest.java

```diff
@@ -105,9 +118,34 @@ public String uri() {
 
     @Override
     public BytesReference content() {
+        assert released.get() == false;
         return content;
     }
 
+    @Override
+    public void release() {
+        if (pooled && released.compareAndSet(false, true)) {
+            request.release();
+        }
+        content = null;
```

Comment:
We don't allow getting `content` after release has been called now -> no point in keeping a reference to a potentially huge object here in case of unpooled buffers.